### PR TITLE
Drop TensorFlow support for Python 3.6

### PR DESCRIPTION
**Pull request by nzw0301**
## Motivation

Tensorflow has dropped Python 3.6 support since its version 2.7.x. Due to its dependencies, sometimes new dependencies cannot be introduced by `optuna/setup.py` for example, `typing-extensions` in #3270. Optuna slowly drops python 3.6 support (https://github.com/optuna/optuna/issues/3021), so it might be a good time to drop Tensorflow's test with Python 3.6 to make Optuna's development easier.

Note that TF 2.6.x, which supports python 3.6, is still developed according to its release note https://github.com/tensorflow/tensorflow/releases/tag/v2.6.3. I'm not sure the future version of 2.6.x will remove the version constraint of `typing-extensions`.

## Description of the changes

- Not Installing TF with python 3.6
- Skip TF-related tests  with python 3.6

### To reviewers

The commit history is a little bit messy. `Squash merge` might be better than the others.

### CODE REVIEW

**Comment by nzw0301** [comment]
Now `typing-extensions-3.10.0.2` can be installed via `pip` (see install section in https://github.com/optuna/optuna/runs/5092511577?check_suite_focus=true). The `typing-extensions` supports `ParamSpec` class that is needed by #3270.

File: setup.py [hunk 1]
```diff
@@ -39,6 +39,7 @@ def get_install_requires() -> List[str]:
         "scipy!=1.4.0" if sys.version[:3] == "3.6" else "scipy>=1.7.0",
         "sqlalchemy>=1.1.0",
         "tqdm",
+        "typing-extensions",
```
Comment (himkt) [inline setup.py:42]:
```suggestion
```

I think `typing-extensions` should be introduced in the PR #3270.

**Review by himkt — COMMENTED** [review]
Thank you so much! Almost LGTM.

**Review by himkt — APPROVED** [review]
LGTM!

**Comment by himkt** [comment]
@toshihikoyanase PTAL.

**Comment by toshihikoyanase** [comment]
@himkt Thank you for letting me know. I'll check it.

**Review by toshihikoyanase — APPROVED** [review]
LGTM! Thank you.

In fact, I was wondering why the `pytorch_lightning` test was ignored, but [this commit](https://github.com/optuna/optuna/pull/3296/commits/bea50ab069c0c3662bfd97c5347571d8048c2b44) reverted it.

**Pull request merged**